### PR TITLE
touch panel to display adjustment

### PR DESCRIPTION
**Pull request by holzingerm**
2nd attempt with a clean version of a touch pane adjustment in case the panel coordinate system do not match the display coordinate.

with this adjustment it is possible to set an offset and a stretch factor for x and y axis of the touch panel.
settings are optional and can be mentioned in advancedsettings.xml

example of my 7inch isp panel (http://www.chalk-elec.com/?p=1712):
<touchscreen_allign>
   
   <x_offset>0</x_offset>
   
   <y_offset>0</y_offset>
   
   <x_stretch_factor>1.44</x_stretch_factor>
   
   <y_stretch_factor>0.89</y_stretch_factor>
</touchscreen_allign>

### CODE REVIEW

File: xbmc/input/MouseStat.h [hunk 1]
```diff
@@ -151,9 +151,9 @@ class CMouseStat
      */
     BUTTON_ACTION Update(unsigned int time, int x, int y, bool down);
   private:
-    static const unsigned int click_confines = 5;        ///< number of pixels that the pointer may move while the button is down to trigger a click
-    static const unsigned int short_click_time = 1000;   ///< time for mouse down/up to trigger a short click rather than a long click
-    static const unsigned int double_click_time = 500;   ///< time for mouse down following a short click to trigger a double click
+//    static const unsigned int click_confines = 8;	///< number of pixels that the pointer may move while the button is down to trigger a click
+    static const unsigned int short_click_time = 1000;   			///< time for mouse down/up to trigger a short click rather than a long click
+    static const unsigned int double_click_time = 500;   			///< time for mouse down following a short click to trigger a double click
```
Comment (AlwinEsch) [inline xbmc/input/MouseStat.h]: Can you remove the tabs here and the click_confines.

File: xbmc/settings/AdvancedSettings.h [hunk 2]
```diff
@@ -344,6 +344,13 @@ class CAdvancedSettings : public ISettingCallback, public ISettingsHandler
     std::string m_cpuTempCmd;
     std::string m_gpuTempCmd;
 
+    // Touchscreen allignment advanced settings
+    int m_xOffset;		// x axis offset of touch panel coordinate origin
+    int m_yOffset;		// y axis offset of touch panel coordinate origin
+    float m_xStretchFactor;	// x axis stretch factor
+    float m_yStretchFactor;	// y axis stretch factor
+    int m_confines; 		// click or touch confines setting
+
```
Comment (AlwinEsch) [inline xbmc/settings/AdvancedSettings.h]: Better to have a more detailed name about the settings label (is a big list there) maybe "m_screenAlign_xOffset".

Also if on xbmc/input/MouseStat.h the description for m_confines description is "number of pixels that the pointer may move while the button is down to trigger a click" better to use the same here.

File: xbmc/settings/AdvancedSettings.cpp [hunk 3]
```diff
@@ -309,6 +309,13 @@ void CAdvancedSettings::Initialize()
   m_iEdlCommBreakAutowait = 0;             // Off by default
   m_iEdlCommBreakAutowind = 0;             // Off by default
 
+  // Touchscreen default values if no adjustment is necessarry
+  m_xOffset = 0;
+  m_yOffset= 0;
+  m_xStretchFactor = 1.0;
+  m_yStretchFactor = 1.0;
+  m_confines = 8;
```
Comment (AlwinEsch) [inline xbmc/settings/AdvancedSettings.cpp]: Possible to leave the default on 5 for m_confines like before?

**Comment by AlwinEsch** [comment]
Has only looked over the code itself but I'm not sure that the implemantation on LinuxInputDevices.cpp and MouseStat.cpp alone is OK, for me yes but not sure.

**Comment by holzingerm** [comment]
I'm sorry to confuse anybody here.

I'm not that familiar with git at the moment.
I pushed a secound commit including the m_confines that I do not want so far.
To clarify a bit, each of the variables are set to a default value in AdvancedSettings.cpp only if the needs to be adjusted, it has the possibility to be replaced by the value stated in /.kodi/userdata/advancedsettings.xml

I don't intend to make/commit changes with an huge impact to the code like it would be with confines so I will try to cancel the 2nd push request.

thank you for reviewing so far

regards
Mario Holzinger

**Pull request closed without merging**